rudder-webapp/SOURCES/rudder-init

```diff
@@ -192,6 +192,21 @@ function reset_policy_server() {
   echo "127.0.0.1" > /var/rudder/cfengine-community/policy_server.dat
 }
 
+function list_allowed_networks() {
+  network="$1"
+  if [ "${network}" = "auto" ]
+  then
+    # list all ipv4 networks matching each non loopback interface
+    for i in $(LANG=C ip -family inet -oneline address | grep -v " lo " | sed -e 's/.*inet \([0-9.\/]\+\) .*/\1/')
+    do
+      ALLOWEDNETWORK[${#ALLOWEDNETWORK[*]}]="${i}"
+    done
+  else
+    # just use the provided network
+    ALLOWEDNETWORK[${#ALLOWEDNETWORK[*]}]="${network}"
+  fi
+}
```

Comment:
As discussed today, we should add a check for an environment variable and a flag file that restricts this auto function to use /32 IP addresses (we don't want to have NO allowed networks because this would break several things).